app/crud.py

```diff
@@ -239,6 +250,17 @@ def increment_product_price_count(db: Session, product: ProductFull):
     return product
 
 
+def decrement_product_price_count(db: Session, product: ProductFull):
+    """Decrement the price count of a product.
+
+    This is used to keep track of the number of prices linked to a product.
+    """
+    product.price_count -= 1
+    db.commit()
```

Comment:
Regarding the atomicity of the transaction, we should avoid adding db.commit, and only commit at the end of the transaction (in peewee we can use a context manager to take care of this, but I don't know how it works in SQLAlchemy).